assembly/integer/u128.ts

```diff
@@ -729,6 +731,82 @@ export class u128 {
     return q;
   }
 
+  /**
+   * Divides a 128-bit number by a 64-bit divisor.
+   * 
+   * @param divisor - The 64-bit divisor.
+   * @returns An array containing the quotient and remainder.
+   * @throws {RangeError} If the divisor is zero or if there's an integer overflow.
+   * 
+   * @remarks
+   * The function normalizes the dividend and divisor to align their most significant bits. 
+   * Normalization simplifies the division process and ensures accurate results, especially 
+   * when working with large numbers. After the division, the result is denormalized to provide 
+   * the actual quotient and remainder.
+   */
+  div64(divisor: u64): u64[] {
+    const dividendHigh: u64 = this.hi;
+    const dividendLow: u64 = this.lo;
+
+    if (divisor == 0) {
+      throw new RangeError("Division by zero");
+    }
+
+    // Panic if the result is larger than 64 bits.
+    if (divisor <= dividendHigh) {
+      throw new RangeError("Integer overflow");
+    }
+
+    // Handle simple division when the high part of the dividend is 0.
+    if (dividendHigh == 0) {
+      return [dividendLow / divisor, dividendLow % divisor];
+    }
+
+    const shiftAmount: i32 = i32(clz(divisor));
+    const normalizedDivisor = u64SafeShl(divisor, shiftAmount);
+    
+    // Align the lower part of the dividend with the divisor.
+    const normalizedDividendLow: u64 = u64SafeShl(dividendLow, shiftAmount);
+
+    // Align the higher part of the dividend:
+    // - Shift the high part left by the same amount as the divisor.
+    // - Shift the low part right to retain the most significant bits.
+    const normalizedDividendHigh: u64 = u64SafeShl(dividendHigh, shiftAmount) | u64SafeShr(dividendLow, 64 - shiftAmount);
+
+    const response = longDivision128by64(normalizedDividendHigh, normalizedDividendLow, normalizedDivisor);
+
+    return [response[0], u64SafeShr(response[1], shiftAmount)];
+  }
+  
+  /**
+   * Returns the quotient and remainder of dividing a 128-bit number by a 64-bit number.
+   * 
+   * @param divisor - The 64-bit divisor.
+   * @returns An array containing the quotient and remainder, both as u128.
+   * 
+   * @remarks
+   * The function uses the long division principle to divide a 128-bit dividend by a 64-bit divisor.
+   * The high and low 64-bit parts of the 128-bit dividend are divided separately by the 64-bit divisor.
+   */
+  quoRem(divisor: u64): u128[] {
```

Comment:
Public `quoRem` is not necessary. See how all this done for `u128.div` and `u128.rem`. I use a common private routine for all of this here: https://github.com/MaxGraey/as-bignum/blob/master/assembly/globals.ts#L248
which later reuse in `u128` and `i128` here: https://github.com/MaxGraey/as-bignum/blob/master/assembly/integer/u128.ts#L411 and https://github.com/MaxGraey/as-bignum/blob/master/assembly/integer/u128.ts#L418

Also I avoid return tuple and use globals like `__divmod_quot_hi`, `__divmod_rem_lo`, `__divmod_rem_hi`. This avoids GC allocs and improve performance

Reply:
Yes, I think `quoRem`, `div64` and `div128` are not necessary. Will be great to have only `div(u256, u256)` and `rem(u256, u256)` which also have overloaded operators.

Reply:
Ok, but using __udivmod128 instead of div64, modified to match your concerns of using the stack and only exposing div and rem, will certainly be less performant.